cloudbuild/presubmit.sh

```diff
@@ -47,9 +47,18 @@ initialize_git_repo() {
 # to determine all changed files and looks for tests in directories with changed files.
 determine_tests_to_run() {
   # Infer the files that changed
+  mapfile -t DELETED_BUILD_FILES < <(git diff origin/master --name-only --diff-filter=D | grep BUILD)
   mapfile -t CHANGED_FILES < <(git diff origin/master --name-only)
+  echo "Deleted BUILD files: ${DELETED_BUILD_FILES[*]}"
   echo "Changed files: ${CHANGED_FILES[*]}"
 
+  # Run all tests if common directories modified by deleting files
+  if [[ -n "${DELETED_BUILD_FILES}" ]]; then
+    echo "All tests will be run: the following BUILD files '${DELETED_BUILD_FILES}' were removed"
```

Comment:
nit: use `${DELETED_BUILD_FILES[*]}` to print all files.

Reply:
Done